### ISSUE TITLE
Add Learning Groups

### ISSUE THREAD

**Issue by Otto-AA** [cardwiki/cardwiki#4]
In some cases it would be helpful to learn several decks together. For instance when learning vocabulary, one might want to mix "Basics", "Travel" and "Phrases" instead of learning them separately.

### Suggested implementation

In the frontend the Learning dashboard has a configurations button, where the user can create learning groups and add decks to it.

The backend stores them as a OneToMany `Deck <-> LearnGroup` relationship. LearnGroup consists of `id`, `decks` and `createdBy`. Users have a new field `learnGroups`.

**Comment by Gittenburg** [comment]
I agree that this would be a good addition. I feel like this should replace our current favorite system (so when you hit save it asks you in what group you want to save the deck). When creating a new user we probably want to start them out with a default group.
